x/supplier/keeper/msg_server_stake_supplier_test.go

```diff
@@ -155,22 +148,124 @@ func TestMsgServer_StakeSupplier_FailWithNonExistingService(t *testing.T) {
 
 	// Stake the supplier & verify that it fails because the service does not exist.
 	_, err := srv.StakeSupplier(ctx, stakeMsg)
 	require.ErrorIs(t, err, types.ErrSupplierServiceNotFound)
 }
 
+func TestMsgServer_StakeSupplier_OperatorAuthorizations(t *testing.T) {
+	k, ctx := keepertest.SupplierKeeper(t)
+	srv := keeper.NewMsgServerImpl(*k.Keeper)
+
+	// Generate an owner and operator address for the supplier
+	ownerAddr := sample.AccAddress()
+	operatorAddr := sample.AccAddress()
+
+	// Stake using the operator address as the signer and verify that it succeeds.
+	stakeMsg := stakeSupplierForServicesMsg(ownerAddr, operatorAddr, 100, "svcId")
+	stakeMsg.Signer = operatorAddr
+	_, err := srv.StakeSupplier(ctx, stakeMsg)
+	require.NoError(t, err)
+	supplier, isSupplierFound := k.GetSupplier(ctx, operatorAddr)
+	require.True(t, isSupplierFound)
+	require.Equal(t, operatorAddr, supplier.Address)
+	require.Equal(t, ownerAddr, supplier.OwnerAddress)
+
+	// Update the supplier using the operator address as the signer and verify that it succeeds.
+	stakeMsg = stakeSupplierForServicesMsg(ownerAddr, operatorAddr, 200, "svcId")
+	stakeMsg.Signer = operatorAddr
+	newUrl := "http://localhost:8081"
+	stakeMsg.Services[0].Endpoints[0].Url = newUrl
+	_, err = srv.StakeSupplier(ctx, stakeMsg)
+	require.NoError(t, err)
+
+	// Check that the supplier was updated
+	foundSupplier, supplierFound := k.GetSupplier(ctx, operatorAddr)
+	require.True(t, supplierFound)
+	require.Equal(t, newUrl, foundSupplier.Services[0].Endpoints[0].Url)
+
+	// Update the supplier using the owner address as the singer and verify that it succeeds.
+	stakeMsg = stakeSupplierForServicesMsg(ownerAddr, operatorAddr, 300, "svcId")
+	newUrl = "http://localhost:8082"
+	stakeMsg.Services[0].Endpoints[0].Url = newUrl
+	_, err = srv.StakeSupplier(ctx, stakeMsg)
+	require.NoError(t, err)
+
+	// Check that the supplier was updated
+	foundSupplier, supplierFound = k.GetSupplier(ctx, operatorAddr)
+	require.True(t, supplierFound)
+	require.Equal(t, newUrl, foundSupplier.Services[0].Endpoints[0].Url)
+
+	// Try updating the supplier's operator address using the owner as a signer
+	// and verify that it fails.
+	stakeMsg = stakeSupplierForServicesMsg(ownerAddr, operatorAddr, 400, "svcId")
+	stakeMsg.Signer = operatorAddr
+	stakeMsg.Address = sample.AccAddress()
+	_, err = srv.StakeSupplier(ctx, stakeMsg)
+	require.NoError(t, err)
+
+	// Check that the supplier was created
+	newSupplier, newSupplierFound := k.GetSupplier(ctx, stakeMsg.Address)
+	require.True(t, newSupplierFound)
+
+	oldSupplier, oldSupplierFound := k.GetSupplier(ctx, operatorAddr)
+	require.True(t, oldSupplierFound)
+
+	require.NotEqual(t, oldSupplier.Address, newSupplier.Address)
+
+	// Trying to update the supplier's owner address using the owner as a signer
+	// will create a new supplier
+	stakeMsg = stakeSupplierForServicesMsg(ownerAddr, operatorAddr, 400, "svcId")
+	stakeMsg.Address = sample.AccAddress()
```

Comment:
1. Ditto regarding the clarity of what the stakeMsg is.
2. Shouldn't this be setting `OwnerAddress` and not `Address`?

Confusing w.r.t the comment above.